### PR TITLE
*: make errcheck work again

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

errcheck is moved to `check_slow` in https://github.com/pingcap/tidb/pull/7240 and it is not used since then.
As the title says, this PR recover the check, and now `make check` will check it by default.

### What is changed and how it works?

- Add errcheck to Makefile
- Fix the unchecked errors

Use errcheck rather than the gometalint (it actually integrate errcheck) tool, because this way we can specify the errcheck_excludes.txt file to control what to be ignored.

### Check List 

Tests 

 - No code

Side effects

There is a known issue errcheck doesn't support module https://github.com/kisielk/errcheck/issues/155
And it introduce a problem that TiDB must be put in GOPATH

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8795)

### CODE REVIEW

File: planner/core/rule_join_elimination.go [hunk 1]
```diff
@@ -74,7 +75,8 @@ func (o *outerJoinEliminator) isAggColsAllFromOuterTable(outerPlan LogicalPlan,
 	}
 	for _, col := range aggCols {
 		columnName := &ast.ColumnName{Schema: col.DBName, Table: col.TblName, Name: col.ColName}
-		if c, _ := outerPlan.Schema().FindColumn(columnName); c == nil {
+		if c, err := outerPlan.Schema().FindColumn(columnName); c == nil {
+			log.Debug(err)
```
Comment (jackysp) [inline planner/core/rule_join_elimination.go]: I think debug level is too low.

**Comment by tiancaiamao** [comment]
/run-all-tests
@winkyao @jackysp

File: planner/core/rule_join_elimination.go (hunk 1 above)
Comment (tiancaiamao) [reply planner/core/rule_join_elimination.go]: This error could be ignored @eurekaka

Comment (zimulala) [reply planner/core/rule_join_elimination.go]: This log is too simple.

File: planner/cascades/optimize.go [hunk 2]
```diff
@@ -73,7 +74,9 @@ func exploreGroup(g *memo.Group) error {
 		// Explore child groups firstly.
 		curExpr.Explored = true
 		for _, childGroup := range curExpr.Children {
-			exploreGroup(childGroup)
+			if err := exploreGroup(childGroup); err != nil {
+				return errors.Trace(err)
```
[the remark this answers is not in the document]
Comment (tiancaiamao) [reply planner/cascades/optimize.go]: https://github.com/pingcap/errors/blob/master/juju_adaptor.go#L27
If an error is traced already, `errors.Trace(err)` will return the original error @winoros

File: planner/core/rule_join_elimination.go (hunk 1 above)
Comment (tiancaiamao) [reply planner/core/rule_join_elimination.go]: Done

File: planner/cascades/optimize.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (winoros) [reply planner/cascades/optimize.go]: But the reason why we use it is just to reduce the call of `errors.Trace`
/cc @lysu

File: executor/builder.go [hunk 3]
```diff
@@ -513,7 +513,9 @@ func (b *executorBuilder) buildShow(v *plannercore.Show) Executor {
 	}
 	if e.Tp == ast.ShowMasterStatus {
 		// show master status need start ts.
-		e.ctx.Txn(true)
+		if _, err := e.ctx.Txn(true); err != nil {
+			b.err = errors.Trace(err)
```
[the remark this answers is not in the document]
Comment (tiancaiamao) [reply executor/builder.go]: The error stack is critical here, I'd rather add more `errors.Trace`!

File: planner/cascades/optimize.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (lysu) [reply planner/cascades/optimize.go]: +1, Trace cannot be inline, so it's better not trace twice in new code

[the remark this answers is not in the document]
Comment (tiancaiamao) [reply planner/cascades/optimize.go]: gotcha

**Comment by tiancaiamao** [comment]
PTAL @winoros @lysu

**Pull request merged**